### PR TITLE
Make filtering consistent with all article functions

### PR DESCRIPTION
**Pull request by kidzegeye**
## Overview

Add Article filter to all Article repo functions that return articles

## Changes Made

- Added article filtering to getArticlesByPublicationSlug and getArticlesByPublicationSlugs
- Turned off article filtering
- Removed getArticlesByPublicationID and getArticlesByPublicationIDs to be consistent with https://github.com/cuappdev/volume-backend/pull/54
- Made IS_FILTER_ACTIVE an environment variable

## Test Coverage

Tested getArticlesByPublicationSlug and getArticlesByPublicationSlugs with articles that were flagged as profane and were not flagged as profane to see if the articles would be filtered correctly

### CODE REVIEW

File: src/common/constants.ts [hunk 1]
```diff
@@ -15,5 +15,5 @@ export const FILTERED_WORDS = [
 ];
 export const IMAGE_ADDRESS = 'https://raw.githubusercontent.com/cuappdev/assets/master/volume';
 export const IOS = 'IOS';
-export const IS_FILTER_ACTIVE = true;
+export const IS_FILTER_ACTIVE = false;
```
Comment (chalo2000) [inline src/common/constants.ts:18]: ironic that the branch is called filter articles but filtering is off

precisely why I said to make this an environment variable in your previous commit - consider making an issue to do this in the future in a small PR

**Review by chalo2000 — APPROVED** [review]
LGTM - change the PR title

File: src/repos/ArticleRepo.ts [hunk 2]
```diff
@@ -96,21 +97,6 @@ const getArticlesByPublicationID = async (
...
-  const uniquePubIDs = [...new Set(publicationIDs)].map((id) => new ObjectId(id));
-  console.log(uniquePubIDs);
-  const pubSlugs = await PublicationModel.find({ _id: { $in: uniquePubIDs } }).select('slug');
-  return getArticlesByPublicationSlugs(
-    pubSlugs.map((pub) => pub.slug),
-    limit,
-    offset,
-  );
-};
-
```
Comment (Archit404Error) [inline src/repos/ArticleRepo.ts:113]: Are we 100% sure the existing iOS app does not make use of this?

**Review by Archit404Error — COMMENTED** [review]
This looks good but not sure why we're deleting `getArticlesByPublicationIDs` in this PR, see comment below

File: src/repos/ArticleRepo.ts [hunk 3]
```diff
@@ -96,21 +97,6 @@ const getArticlesByPublicationID = async (
...
-): Promise<Article[]> => {
-  const uniquePubIDs = [...new Set(publicationIDs)].map((id) => new ObjectId(id));
-  console.log(uniquePubIDs);
-  const pubSlugs = await PublicationModel.find({ _id: { $in: uniquePubIDs } }).select('slug');
-  return getArticlesByPublicationSlugs(
-    pubSlugs.map((pub) => pub.slug),
-    limit,
-    offset,
-  );
-};
```
Comment (Archit404Error) [inline src/repos/ArticleRepo.ts:112]: Are we 100% sure the existing iOS app does not make use of this?

Comment (kidzegeye) [reply src/repos/ArticleRepo.ts:112]: Will readd, I thought it was removed in main

File: .envrctemplate [hunk 4]
```diff
@@ -1,3 +1,4 @@
 export NODE_ENV=[node env]
 export APP_PORT=[port]
-export MONGO_ADDRESS=[db uri]
\ No newline at end of file
+export MONGO_ADDRESS=[db uri]
+export IS_FILTER_ACTIVE = false
```
Comment (chalo2000) [inline .envrctemplate:4]: the template should not hold the actual value, I would replace these with
```
export MONGO_ADDRESS=[mongo address]
export IS_FILTER_ACTIVE=[is filter active]
```

Comment (chalo2000) [reply .envrctemplate:4]: also include that you added environment variables in the PR description

Comment (chalo2000) [reply .envrctemplate:4]: I also see MONGO_ADDRESS isn't used anywhere in the code for this PR?

**Review by chalo2000 — APPROVED** [review]
review my comments and merge if nothing pressing

**Review by Archit404Error — APPROVED** [review]
Agree with Gonzalo's comments, good to go after that

**Pull request merged**